chat.delta.desktop.yml

```diff
@@ -112,45 +110,64 @@ modules:
   - name: delta
     buildsystem: simple
     build-options:
-      append-path: /usr/lib/sdk/node20/bin
+      append-path: /usr/lib/sdk/node20/bin:/run/build/delta/pnpm/bin
       env:      
         NPM_CONFIG_LOGLEVEL: info
 
-        # Tell caches where we pre-downloaded things.
-        npm_config_cache: /run/build/delta/flatpak-node/npm-cache/
-
-        # for node gyp:
-        npm_config_nodedir: /run/build/delta/node-headers
-
         XDG_CACHE_HOME: /run/build/delta/flatpak-node/cache
         TMPDIR: /run/build/delta/flatpak-node/tmp
 
         # This overwrites the git-ref version info for the about- and crash-screen
         VERSION_INFO_GIT_REF: "flathub"
         NO_ASAR: "true"
 
+        ELECTRON_CONFIG_CACHE: /run/build/delta/cache/electron
+        electron_config_cache: /run/build/delta/cache/electron
+        ELECTRON_CACHE: /run/build/delta/cache/electron
+
     build-commands:
+      - pwd
+      - ls
+      # fixup manually installed pnpm package (why manual? because global npm install is not possible because global node modules dir is read-only)
+      - mv ../pnpm/bin/pnpm.cjs ../pnpm/bin/pnpm
       - ls -lah /run/build/
       - mkdir -p $TMPDIR
       - node --version
-      - npm --version
+      - pnpm --version
       - python --version
-      - npm cache verify
-      # https://github.com/flatpak/flatpak-builder-tools/blob/db24c70456b7122f4243ab4a21dbb3368334e259/node/README.md#chromedriver-support
-      - env DEBUG="*" npm install --offline --verbose
+      - pnpm config set registry http://localhost:3000 --location project
       # install the core we build locally instead of the prebuild
-      - env DEBUG="*" npm install --install-links=true --offline --verbose /app/dc-core/deltachat-rpc-server/npm-package
-      - rm -r /app/dc-core
-      - npm run build4production --offline --verbose
-      - npm run pack:generate_config --offline --verbose
+      # patch package.json to exclude the extra architectures that we excluded when building the pnpm store
+      - |
+        jq ".pnpm.supportedArchitectures.os = [\"linux\"] | .pnpm.supportedArchitectures.cpu = [\"x64\",\"arm64\"]" package.json > package.new.json
+        mv package.new.json package.json
+      # install deps from store
       - |
```

Comment:
Just for understanding: why is this needed?  Are the dc core binaries not already downloaded in the previous build-commands?

Reply:
it excludes the architectures from installation that we didn't include in the cache, because they are not used.

pnpm downloads all architectures for dc core currently when you do pnpm install.